nodePackages.wavedrom-cli: init at 2.6.9

expipiplus1 opened the pull request:
Also update nodePackages

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

expipiplus1 committed:
nodePackages.wavedrom-cli: init at 2.6.9

Also update nodePackages

r-rmcgibbo commented:
Result of `nixpkgs-review pr 123065` at c4e2c673 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/e709a6cdb6e520716e2317260a964533">python39Packages.batchspawner</a></li>
    <li><a href="https://gist.github.com/f981c9e12cc69ccd18eaccccbef9e0e1">python39Packages.dockerspawner</a></li>
    <li><a href="https://gist.github.com/7d4712df3b89ed4d3fb594e0575f8c7f">python39Packages.jupyterhub</a></li>
    <li><a href="https://gist.github.com/ae2a25b9fb7e3852b31840e0e7077209">python39Packages.jupyterhub-ldapauthenticator</a></li>
    <li><a href="https://gist.github.com/a298d42d9896ac02e9dd5085965cc7f7">python39Packages.jupyterhub-systemdspawner</a></li>
    <li><a href="https://gist.github.com/82af21b04dc8a53b7e075f50c4813d5c">python39Packages.jupyterhub-tmpauthenticator</a></li>
    <li><a href="https://gist.github.com/dd0e97bc517a2172844beb1554408e92">python39Packages.oauthenticator</a></li>
  </ul>
</details>
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>iosevka</li>
    <li>lumo</li>
  </ul>
</details>
<details>
  <summary>15 packages built successfully:</summary>
  <ul>
    <li>bat-extras.prettybat</li>
    <li>discourse</li>
    <li>mx-puppet-discord</li>
    <li>parity-ui</li>
    <li>python38Packages.batchspawner</li>
    <li>python38Packages.dockerspawner</li>
    <li>python38Packages.jupyterhub</li>
    <li>python38Packages.jupyterhub-ldapauthenticator</li>
    <li>python38Packages.jupyterhub-systemdspawner</li>
    <li>python38Packages.jupyterhub-tmpauthenticator</li>
    <li>python38Packages.oauthenticator</li>
    <li>teleprompter</li>
    <li>vscode-extensions.matklad.rust-analyzer</li>
    <li>vscode-extensions.vadimcn.vscode-lldb</li>
    <li>zerobin</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

expipiplus1 commented:
Does anyone with more nix node knowledge know what do do about that failure?

midchildan commented:
@expipiplus1 They all seem to be unrelated failures. The netlify-cli failure appears to be caused by #122498, but that should probably be addressed in another PR.

midchildan commented:
The Darwin build for this package is failing because node-gyp is attempting to check for the existence of the XCode Command Line Tools. Nixpkgs seems to have a [patch][1] for this, but it's no longer working as intended.

This can probably go to another PR too.

Related: #102973

[1]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/web/nodejs/bypass-xcodebuild.diff

midchildan commented on pkgs/development/node-packages/default.nix:
```diff
@@ -326,5 +326,14 @@ let
+
+    wavedrom-cli = super.wavedrom-cli.override {
+      nativeBuildInputs = [ pkgs.pkg-config self.node-pre-gyp ];
+      buildInputs = with pkgs; [
```
It would helpful to add a comment explaining that the dependencies are required by https://github.com/Automattic/node-canvas. It took a bit of digging for me to find out.

SuperSandro2000 replied:
This should be moved into node-canvas that other packages using it benefit from it, too.

expipiplus1 replied:
perhaps we could leave that to a separate PR

midchildan replied:
@SuperSandro2000 IIRC there's no easy way to fix intermediate dependencies with node2nix, so the easiest way to fix them would be in the leaf packages. I might be mistaken though.

expipiplus1 commented:
> The Darwin build for this package is failing because node-gyp is attempting to check for the existence of the XCode Command Line Tools. Nixpkgs seems to have a [patch](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/web/nodejs/bypass-xcodebuild.diff) for this, but it's no longer working as intended.

Probably not appropriate to mark this as Linux only though.

> @expipiplus1 They all seem to be unrelated failures.

Great!

I've added the requested comment and rebased. Thanks for the review, @midchildan

midchildan commented:
Change looks good. It needs a rebase and a re-run of `./generate.sh` to fix the merge conflict.

expipiplus1 commented:
> Change looks good. It needs a rebase and a re-run of `./generate.sh` to fix the merge conflict.

done, thanks!

expipiplus1 commented:
thanks！

On Thu, 3 Jun 2021, 10:51 pm Sandro, ***@***.***> wrote:

> Merged #123065 <https://github.com/NixOS/nixpkgs/pull/123065> into master.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/123065#event-4837997613>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGRJXBOCJDIR77JXKCLSELTQ6H3RANCNFSM445RS5BQ>
> .
>
